taichi/program/kernel.h

```diff
@@ -41,6 +41,13 @@ class Kernel : public Callable {
                                 uint64 size,
                                 bool is_device_allocation);
 
+    void set_arg_external_array_w_shape(int arg_id,
+                                        uint64 ptr,
```

Comment:
```suggestion
                                        uintptr_t ptr,
```

Reply:
I used a consistent interface with `set_arg_external_array`, which takes `uint64` as type identifier for ptr. Shall we change them all?

Reply:
SG, thx!